SE_Topics.md

```diff
@@ -9,13 +9,17 @@ Please feel free to submit a PR with suggestions of topics to add.
 * Version control
 * Automated testing
 * Code review
-* Specification Writing
-* Code smells, naming practices, code style etc ?
+* Requirements Engineering
+* Code smells (perhaps merge with source code static analysis)
+* Naming practices, code style etc 
 * Bug reporting: how to write a good bug report ?
 * Pair programming
-
-
-
+* Software Modeling (how much should we model? using what? when (what kind of projects)? etc.)
+* Software Design/Architecture
+* Software Effort Estimation
```

Comment:
I honestly wouldn't feel bad if we didn't get into this - it's at least as much an art as a science and is going to be wildly different in different industrial domains.

Reply:
I think it is important that students develop the ability to criticise different architecture styles rather than reproducing the information they get. For instance, why using REST architecture combined with Pipes and Filters to solve problem A? Why microservices? What are the benefits and drawbacks of MVC? Why solution A scales better than B? Why team A is using Firebase and team B decided to use Mongo for the same kind of product? etc.

In terms of software design, what is a good and bad design for certain problems? Not to mention software modularity issues and design patterns.

With respect to software effort estimation, do the software engineers we're graduating know how to estimate effort well? Do they know what are the options to estimate effort? Do they practice in college before jumping into a new project?

I agree that some topics seem to be as much an art as a science, but the main point is: do we see these topics as relevant topics for the students to learn (regardless art vs science)? For example: do software engineers have to estimate effort? If so, is it important to provide the students the chance to learn this in our classes?  Do engineers need to think of an upfront architecture/design? If so, how can we have it in our classes?

Reply:
Oh, I didn't mean design, I just meant estimation.  I think it'd be *hard* to teach design, but awesome if we could.

As far as estimation goes, I've had nearly no positive experiences with estimation techniques. I'd approach the whole thing very skeptically without hearing some very very positive first-hand experiences from practitioners.

Reply:
I have positive and negative experiences with effort estimation (from industry and academia). I'd like to hear more opinions too.

Reply:
I think effort estimation is important and maybe interesting for students, though I miss any experience in teaching that.
However, it may be that, at the end, we have to many topics/modules and this may be skipped in an introductory SE class.
Nevertheless, I think it is worth to keep. But it should include all phases (e.g., also the effort for maintenance)

Reply:
Maybe pull it out of "Core topics" at least?